state/futures/selectors.ts

```diff
@@ -1,47 +1,76 @@
 import { createSelector } from '@reduxjs/toolkit';
 import { wei } from '@synthetixio/wei';
-import { selectExchangeRatesWei } from 'state/exchange/selectors';
-import { deserializeWeiObject } from 'state/helpers';
-import { RootState } from 'state/store';
 
-import { FuturesMarket, FuturesPosition } from 'queries/futures/types';
+import { FuturesPosition } from 'queries/futures/types';
 import { PositionSide } from 'sections/futures/types';
+import { selectExchangeRates } from 'state/exchange/selectors';
+import { accountType, deserializeWeiObject } from 'state/helpers';
+import { RootState } from 'state/store';
 import { newGetExchangeRatesForCurrencies } from 'utils/currencies';
 import { zeroBN } from 'utils/formatters/number';
-import { MarketKeyByAsset } from 'utils/futures';
+import { MarketKeyByAsset, unserializeFundingRates, unserializeMarkets } from 'utils/futures';
+
+import { FundingRate } from './types';
 
 export const selectMarketKey = createSelector(
-	(state: RootState) => state.futures.marketAsset,
+	(state: RootState) => state.futures[accountType(state.futures.selectedType)].marketAsset,
 	(marketAsset) => MarketKeyByAsset[marketAsset]
 );
 
+export const selectFuturesType = (state: RootState) => state.futures.selectedType;
+
+export const selectMarketAsset = createSelector(
+	(state: RootState) => state.futures,
+	selectFuturesType,
+	(futures, marginType) => futures[accountType(marginType)].marketAsset
+);
+
+export const selectMarketRate = createSelector(
+	selectMarketKey,
+	selectExchangeRates,
+	(marketKey, exchangeRates) => newGetExchangeRatesForCurrencies(exchangeRates, marketKey, 'sUSD')
+);
+
+export const selectMarkets = (state: RootState) => unserializeMarkets(state.futures.markets);
```

Comment:
Looks like you might have skipped this selector by accident. Adding this suggestion, to help make the change a one-click one:

```suggestion
export const selectMarkets = createSelector(
  (state: RootState) => state.futures.markets,
  unserializeMarkets
);
```